src/plugins/condition/components/Criterion.vue

```diff
@@ -178,17 +179,17 @@ export default {
     methods: {
         checkTelemetry() {
             if(this.criterion.telemetry) {
-                if (this.criterion.telemetry === 'any' || this.criterion.telemetry === 'all') {
-                    this.updateMetadataOptions();
+                const isAnyAllTelemetry = this.criterion.telemetry === 'any' || this.criterion.telemetry === 'all';
+                if (!isAnyAllTelemetry &&
+                    !this.telemetry.find((telemetryObj) => this.openmct.objects.areIdsEqual(this.criterion.telemetry, telemetryObj.identifier))) {
```

Comment:
This feels like it should be in a const, like `isAnyAllTelemetry`. Personally, as a developer, I'm stupid and lazy (also usually tired), so just tell me what this code does with a descriptive name so I don't have to figure it out.

Reply:
Done.